### PR TITLE
feat: nisar sim v2

### PR DESCRIPTION
**Pull request by jjfrench**
Before you submit a pull request, please fill in the following:

**Description:**
Slimming down the NISAR-sim package to reflect the latest simulated NISAR data.

Removed:
- requirements-dev.txt
- setup.cfg

Modified:
- pyproject.toml 
  - moved everything over from the setup.cfg and updated some of the configurations such as using ruff and some of the dependencies.
  - ensures that some packages get installed by listing them in the dependencies here
- Dockerfile
  - consolidated down to one virtual environment, when trying to run the commands I kept seeing that packages were missing but the base environment had most of the packages installed that I needed. The commands were being triggered from one environment and the installs were happening in another

This latest simulated NISAR package is based on the latest simulated NISAR product suite: https://nisar.jpl.nasa.gov/data/sample-data/, where there are 10 different data products with relatively the same h5 structure (with the exception of Soil Moisture). 

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md). (Was never released - No changes)

### CODE REVIEW

**Review by omshinde — APPROVED** [review]
Looks good to me. Thank you for your efforts.

1. Probably for later versions, we might need to add properties from here: https://github.com/stac-extensions/sar. Specifically, `product type`, `frequency`, etc.  
2. Also, is `RRSD` a good key name for a general audience? I am not sure if the asset names should target a general audience or domain scientists but `RRSD` is difficult to interpret IMO.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -14,7 +14,19 @@
...
-  
+    // Attempting to add mounts to AWS credentials for development environment
+    "mounts": [
+      {
+        "source": "${localEnv:HOME}/.aws",
+        "target": "/root/.aws",
+        "type": "bind"
+      },
+      {
+        "source": "${localEnv:HOME}/.ssh",
```
Comment (emileten) [inline .devcontainer/devcontainer.json:25]: what were you using your SSH configuration for ?

File: .github/PULL_REQUEST_TEMPLATE.MD [hunk 2]
```diff
@@ -1,16 +0,0 @@
-Before you submit a pull request, please fill in the following:
```
Comment (emileten) [inline .github/PULL_REQUEST_TEMPLATE.MD:1]: Why are we removing the PR template?

File: docker/Dockerfile [hunk 3]
```diff
@@ -8,51 +8,46 @@ RUN conda update conda && \
 
 FROM base as dependencies
 
-ENV PATH="/opt/venv/bin:$PATH"
```
Comment (emileten) [inline docker/Dockerfile:11]: These dockerfile changes make sense to me, but I am surprised it was even working with the previous version of the Dockerfile ? 🤔

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,66 @@
+[project]
+name = "stactools-nisar-sim"
```
Comment (emileten) [inline pyproject.toml:2]: There is quite a lot of changes in terms of how we install dependencies and manage environments in this PR: `pyproject.toml`, `requirements-dev.txt`, the `Dockerfile`, `setup.cfg`. I remember indeed struggling with environment set up when I worked on V1. 

Could you provide a brief explanation of these changes in the PR description ?

**Review by emileten — COMMENTED** [review]
Thanks for the PR @jjfrench 

I have a couple questions that came up while reading this. Maybe they could be briefly addressed if you provide a brief description of the changes made in this PR. I see two general changes, that you could maybe shed light on with just a couple lines in the description.

- dependencies management (see my comments)
- application changes (changes to `src/stactools/nisar_sim` and related changes in `tests`)

Thanks !

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (jjfrench) [reply .devcontainer/devcontainer.json:25]: It was a copied configuration, it doesn't really matter

File: .github/PULL_REQUEST_TEMPLATE.MD (hunk 2 above)
Comment (jjfrench) [reply .github/PULL_REQUEST_TEMPLATE.MD:1]: The most recent stactools template removed it

File: docker/Dockerfile (hunk 3 above)
Comment (jjfrench) [reply docker/Dockerfile:11]: I had installed the needed packages into the separate build stages in the previous dockerfile which was "nicht gute"

File: pyproject.toml (hunk 4 above)
Comment (jjfrench) [reply pyproject.toml:2]: I'll cross-post it here for visibility.

Removed:
- requirements-dev.txt
- setup.cfg
Modified:
- pyproject.toml 
  - moved everything over from the setup.cfg and updated some of the configurations such as using ruff and some of the dependencies.
  - ensures that some packages get installed by listing them in the dependencies here
- Dockerfile
  - consolidated down to one virtual environment, when trying to run the commands I kept seeing that packages were missing but the base environment had most of the packages installed that I needed. The commands were being triggered from one environment and the installs were happening in another

Comment (jjfrench) [reply pyproject.toml:2]: (I didn't want to necessarily add these changes to the changelog, since the package was never released)

**Comment by jjfrench** [comment]
Thanks for the feedback @emileten, let me know if you have any other questions! Feel free to resolve conversations if I was able to answer your question.

File: .github/PULL_REQUEST_TEMPLATE.MD (hunk 2 above)
Comment (jjfrench) [reply .github/PULL_REQUEST_TEMPLATE.MD:1]: It was also annoying because it was failing linting 😛

Comment (jjfrench) [reply .github/PULL_REQUEST_TEMPLATE.MD:1]: I takeback what I said previously, they didn't update new templates

**Comment by emileten** [comment]
Thanks so much for the explanations @jjfrench! Feel free to merge, I resolved the conversations.

**Review by emileten — APPROVED** [review]

**Pull request merged**